eitprocessing/datahandling/mixins/equality.py

```diff
@@ -13,9 +14,10 @@ class Equivalence:
     """Mixin class that adds an equality and equivalence check."""
 
     # inspired by: https://stackoverflow.com/a/51743960/5170442
-    def __eq__(self, other: Self):
+    def __eq__(self, other: Self) -> bool:
         if self is other:
             return True
+
         if is_dataclass(self):
             if self.__class__ is not other.__class__:
                 return NotImplemented
```

Comment:
I think raising `NotImplemented` should happen whenever classes of two objects don't match, not only if we're dealing with data classes.
Also, what is the reason to check the `__class__` instead of the `type()` (as done in line 81)?

```suggestion
        if self is other:
            return True
        if self.__class__ is not other.__class__:
            return NotImplemented

        if is_dataclass(self):
```

Reply:
I think `NotImplemented` is not the right result here anyway. If the types don't match, dataclass or not, `__eq__` should return False. Or am I missing something?

I changed it to return `False` in a968c9daa4bf22c36b616a310d9b0c8e69b8436d.